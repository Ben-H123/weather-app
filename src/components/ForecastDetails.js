/* eslint-disable prettier/prettier */
import React from "react";

function ForecastDetails({forecast}) {
  const {
    date,
    temperature:{min, max},
    humidity,
    wind:{speed, direction},
  } = forecast;

  const formattedDate = new Date(date).toDateString()

  return (
    <div className="forecast-details">
    <div className="forecast-details__date">
      {formattedDate}
    </div>
    <div className="forecast-details__max-temperature">
      {max}
    </div>
    <div className="forecast-details__min-temperature">
      {min}
    </div>
    <div className="forecast-details__humidity">
      {humidity}
    </div>
    <div className="forecast-details__wind-speed">
      {speed}
    </div>
    <div className="forecast-details__wind-direction">
      {direction}
    </div>
  </div>     
      )
  };

  export default ForecastDetails;