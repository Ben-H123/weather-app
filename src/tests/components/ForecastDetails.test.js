import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from '../../components/ForecastDetails';
import forecast from '../../data/forecast.json';

describe("ForecastDetails", () => {
    const validProps = forecast.forecasts[0];

    it("renders", () => {
    const { asFragment } = render(
      <ForecastDetails
    forecasts = {validProps}
      />
    );

    expect(asFragment()).toMatchSnapshot();
    })
});