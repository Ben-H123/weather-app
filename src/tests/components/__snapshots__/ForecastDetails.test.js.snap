// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon Apr 30 2018
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      11
    </div>
    <div
      class="forecast-details__min-temperature"
    >
      4
    </div>
    <div
      class="forecast-details__humidity"
    >
      30
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      13
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      s
    </div>
  </div>
</DocumentFragment>
`;
